examples/dojo/js/todo/store/TodoLocalStorage.js

```diff
@@ -0,0 +1,156 @@
+		 */
+		postscript: function (props) {
+			lang.mixin(this, props);
+		},
+
+		/**
+		 * Retrieves an object by its identity.
+		 * @param {number} id The identity to use to lookup the object
+		 * @returns {Object} The object in the store that matches the given id.
+		 */
+		get: function (id) {
+			var data = this._loadStorage();
+			var index = findIndex(data, function (item) {
+				return this.getIdentity(item) === id;
+			}, this);
+			if (index >= 0) {
+				return data[index];
+			}
+		},
+
+		/**
+		 * @param {Object} object The object to get the identity from.
+		 * @returns {number} The identify of the given object.
+		 */
+		getIdentity: function (object) {
+			return object[this.idProperty];
+		},
+
+		/**
+		 * Stores an object.
+		 * @param {Object} object The object to store.
+		 * @param {Object} [options]
+		 *     Additional metadata for storing the data.
+		 *     Includes an `id` property if a specific id is to be used.
+		 * @returns {number} The ID of the stored object.
+		 */
+		put: function (object, options) {
+			var id = options && options.id || this.getIdentity(object) || Math.random();
+			var data = this._loadStorage();
+			var index = findIndex(data, function (item) {
+				return this.getIdentity(item) === id;
+			}, this);
+			data[index >= 0 ? index : data.length] = object;
+			object[this.idProperty] = id;
+			this._saveStorage(data);
+			return id;
+		},
+
+		/**
+		 * Creates an object, throws an error if the object already exists.
+		 * @param {Object} object The object to store.
+		 * @param {Object} [options]
+		 *     Additional metadata for storing the data.
+		 *     Includes an `id` property if a specific id is to be used.
+		 */
+		add: function (object, options) {
+			if (this.get(this.getIdentity(object))) {
+				throw new Error('Object already exists.');
+			}
+			return this.put(object, options);
+		},
+
+		/**
+		 * Deletes an object by its identity.
+		 * @param {number} id The identity to use to delete the object.
+		 */
+		remove: function (id) {
+			var data = this._loadStorage();
+			var index = findIndex(data, function (item) {
+				return this.getIdentity(item) === id;
+			}, this);
+			if (index < 0) {
+				throw new Error('Object not found.');
+			}
+			data.splice(index, 1);
+			this._saveStorage(data);
+		},
+
+		/**
+		 * Queries the store for objects.
+		 * @param {Object} query The query to use for retrieving objects from the store.
+		 * @param {dojo/store/util/SimpleQueryEngine.__queryOptions} [options]
+		 *     The optional arguments to apply to the resultset.
+		 * @returns {dojo/store/util/QueryResults} The results of the query, extended with iterative methods.
+		 * @example
+		 * <caption>
+		 *     Find all completed items.
+		 * </caption>
+		 * var results = store.query({completed: true});
+		 */
+		query: function (query, options) {
+			/*jshint newcap:false*/
+			return QueryResults(this.queryEngine(query, options)(this._loadStorage()));
+		}
+	});
+});
```

Comment:
Can you add a TODO comment about removing it when the above is fixed an link to the ticket? So we don't forget.